### PR TITLE
Update to azure-mgmt-sql 0.6.0.

### PR DESCRIPTION
**Pull request by jaredmoo**
The main change is that sql server operations use api-version 2015-05-01-preview instead of 2014-04-01. The new api version is compatible other than one minor difference: administratorLoginPassword is no longer returned in the response to a PUT request.

All tests are re-recorded due to the new server api-version.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by troydai** [comment]
Test tests.test_rp_auto_reg.TestRPAutoRegister need re-recording.
@yugangw-msft

**Comment by jaredmoo** [comment]
I tried to re-record but I can't unregister Microsoft.Sql because all my subscriptions have sql resources that I need. Can one of you guys try recording?

**Comment by yugangw-msft** [comment]
Please just comment out the test. The logic it tested was ported to msrest-azure and published last week. I plan to rip off the production code in near future

File: src/azure-cli-core/tests/test_rp_auto_reg.py [hunk 1]
```diff
@@ -12,23 +12,23 @@
...
+#                 self.cmd('provider unregister -n Microsoft.Sql --wait')
+#                 time.sleep(30)  # a bit random but like to ensure unregistering went through
+
+#         return super(TestRPAutoRegister, self).setUp()
+
+#     @ResourceGroupPreparer()
+#     def test_rp_auto_register(self, resource_group, resource_group_location):
+#         cmd = ('sql server create -g {} -n ygserver123 -l {} --admin-user sa123 --admin-password verySecret12345')
+#         self.cmd(cmd.format(resource_group, resource_group_location), checks=JMESPathCheck('name', 'ygserver123'))
+#         self.cmd('provider show -n Microsoft.Sql', checks=JMESPathCheck('registrationState', 'Registered'))
```
Comment (troydai) [inline src/azure-cli-core/tests/test_rp_auto_reg.py]: Thanks for the change @jaredmoo Actually you can just delete these code. I just confirmed with @yugangw-msft , it's good to go.

**Pull request merged**